web/modules/weather_data/src/Service/DailyForecastTrait.php

```diff
@@ -59,10 +71,17 @@ private function formatDailyPeriodForToday($period, $timezone = null)
             $timezone,
         );
         $endTime = DateTimeUtility::stringToDate($period->endTime, $timezone);
-        $midnight = $startTime->setTime(0, 0);
-        $overnightEnd = $startTime->setTime(6, 0);
-        $isOvernightPeriod =
-            $startTime >= $midnight && $endTime <= $overnightEnd;
+        $midnight = $now->setTime(0, 0);
+        $overnightEnd = $now->setTime(6, 0);
+
+        // This is an overnight period if the current time is after midnight and
+        // the period ends on or before 6am of the same day.
+        //
+        // If now is before midnight, this must either be a day or night period.
+        //
+        // If now is after midnight and this period ends after 6am, then it must
+        // also be a day or night period.
+        $isOvernightPeriod = $now >= $midnight && $endTime <= $overnightEnd;
```

Comment:
I need to think through this some more. I believe as-written, `$now` will by definition always be after `$midnight`. 🤔 Maybe it should be:

```php
$isOvernightPeriod = intval($now->format("G")) <= 6 && $endTime <= $overnightEnd;
```

That is, if it is currently between midnight and 6am and this period ends at or before 6am, then this period must be overnight.

Reply:
So it's not strictly necessary because both the API and our own code should have filtered out the potential overnight period once the actual time reaches 6am, but there's no harm in a little extra defensive coding.